packages/runtime/src/ctx.ts

```diff
@@ -484,7 +489,7 @@ export class Ctx implements Context {
     }
 
     const appParams = this.getApp(appID);
-    this.runtime.run(appParams[APPROVAL_PROGRAM], ExecutionMode.APPLICATION, this.debugStack);
+    this.runtime.run(appParams[APPROVAL_PROGRAM], ExecutionMode.APPLICATION, idx ?? 0, this.debugStack);
```

Comment:
When can `idx` be `undefined`?

Reply:
when user directly calls updateApp or maybe we can handle it in runtime.ts, it will be more clear

Reply:
moved